openssl/src/pkcs7.rs

```diff
@@ -281,11 +296,37 @@ impl Pkcs7Ref {
             Ok(stack)
         }
     }
+
+    // Return the type of a PKCS#7 structure as an Asn1Object
+    pub fn type_nid(&self) -> Option<&Asn1ObjectRef> {
```

Comment:
`type_nid()` seems not ideal for something returning a (reference to an) object identifier. Perhaps `object()` as in `X509AlgorithmRef`, or `type_oid()` or similar?